modules/system/assets/js/framework.js

```diff
@@ -106,9 +108,50 @@ if (window.jQuery.request !== undefined) {
             $.extend(data, options.data)
         }
 
+        // Identify and merge the data from parent form elements
+        //
+        // If the current $form element was spawned by an element that exists in a different form element
+        // (usually because the current form is in a popup), then it will have a data-request-parent-element
+        // attribute present that identifies the element that spawned it. We then need to merge the data
+        // from that element's containing form element and in turn check to see if it was spawned by another
+        // element (i.e. nested popups) until we have completed the chain down to the original form element.
+        // The closer a form element is to the current request $el, the higher priority its data will be in
+        // merge conflicts. This is required in order to ensure that all the data that was required to render
+        // a given widget is still present in any requests made to that widget so that it will be properly
+        // instantiated on the server side and ready to respond to our requests.
+        var parentFormData = {},
+            $parentEl = false,
+            parentForms = [$form],
+            findParentForms = function ($form) {
+                // If the form element exists and has a parent element defined
+                if ($form.length && $form.data('request-parent-element')) {
+                    // Identify the owning form of the parent element
+                    $parentEl = $($form.data('request-parent-element'))
+                    if ($parentEl.length) {
+                        var $parentForm = $parentEl.closest('form')
+                        if ($parentForm.length) {
+                            // Add the identified parent form to the array for processing it's data
+                            // and check it for a parent element & containing form of its own
+                            parentForms.push($parentForm)
+                            findParentForms($parentForm)
+                        }
+                    }
+                }
+            }
+        findParentForms($form)
+        parentForms.reverse().forEach(function ($formEl) {
+            $.extend(parentFormData, serializeFormToObj($formEl))
+        })
```

Comment:
@LukeTowers maybe instead of `$formEl`, it could be `$childForm` just to make it clear what we're dealing with in this loop.
```suggestion
        parentForms.reverse().forEach(function ($childForm) {
            $.extend(parentFormData, serializeFormToObj($childForm))
        })
```

Reply:
I feel like $parentForm is more accurate, I'll change up the variables a bit